src/actions/channels.js

```diff
@@ -690,6 +690,9 @@ export function viewChannel(channelId: string, prevChannelId: string = ''): Acti
         const {myPreferences} = getState().entities.preferences;
         const viewTimePref = myPreferences[`${Preferences.CATEGORY_CHANNEL_APPROXIMATE_VIEW_TIME}--${channelId}`];
         const viewTime = viewTimePref ? parseInt(viewTimePref.value, 10) : 0;
+        const pchanMannuallyUnread = isManuallyUnread(prevChannelId, getState());
+        const preChanId = pchanMannuallyUnread ? '' : prevChannelId;
```

Comment:
Since we can't modify `prevChannelId`, maybe we can just inline this when we call `Client4.viewMyChannel` to avoid two similarly-named variables?
```javascript
await Client4.viewMyChannel(channelId, prevChannelManuallyUnread ? '' : prevChannelId));
```